datasets/lst20/lst20.py

```diff
@@ -0,0 +1,197 @@
+  2. Extract the .tar.gz; this will result in folder `LST20Corpus`
+  The <path/to/folder> can e.g. be `~/Downloads/LST20Corpus`.
+  lst20 can then be loaded using the following command `datasets.load_dataset("lst20", data_dir="<path/to/folder>")`.
+  """
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "fname": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.features.ClassLabel(names=self._POS_TAGS)),
+                    "ner_tags": datasets.Sequence(datasets.features.ClassLabel(names=self._NER_TAGS)),
+                    "clause_tags": datasets.Sequence(datasets.features.ClassLabel(names=self._CLAUSE_TAGS)),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://aiforthai.in.th/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        data_dir = os.path.abspath(os.path.expanduser(dl_manager.manual_dir))
+
+        # check if manual folder exists
+        if not os.path.exists(data_dir):
+            raise FileNotFoundError(
+                f"{data_dir} does not exist. Make sure you insert a manual dir via `datasetts.load_dataset('lst20', data_dir=...)`. Manual download instructions: {self.manual_download_instructions})"
+            )
+
+        # check number of .txt files
+        nb_train = len(glob.glob(f"{data_dir}/train/*.txt"))
+        nb_valid = len(glob.glob(f"{data_dir}/eval/*.txt"))
+        nb_test = len(glob.glob(f"{data_dir}/test/*.txt"))
+        assert (
+            nb_train > 0
+        ), f"No files found in train/*.txt.\nManual download instructions:{self.manual_download_instructions})"
+        assert (
+            nb_valid > 0
+        ), f"No files found in eval/*.txt.\nManual download instructions:{self.manual_download_instructions})"
+        assert (
+            nb_test > 0
+        ), f"No files found in test/*.txt.\nManual download instructions:{self.manual_download_instructions})"
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(data_dir, self._TRAINING_FOLDER)},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"filepath": os.path.join(data_dir, self._VALID_FOLDER)},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, self._TEST_FOLDER)},
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        for fname in glob.glob(f"{filepath}/*.txt"):
```

Comment:
```suggestion
        for fname in sorted(glob.glob(f"{filepath}/*.txt")):
```
we need to make the order deterministic here